src/http/api/views/reply.chain.view.integration.test.ts

```diff
@@ -0,0 +1,168 @@
+import { unsafeUnwrap } from 'core/result';
+import type { Knex } from 'knex';
+import type { Post } from 'post/post.entity';
+import { createTestDb } from 'test/db';
+import { type FixtureManager, createFixtureManager } from 'test/fixtures';
+import { beforeAll, beforeEach, describe, expect, it } from 'vitest';
+import { ReplyChainView } from './reply.chain.view';
+
+/**
+ * This will setup the database with a bunch of posts centered around a single post.
+ * The post will have an ancestor chain of MAX_ANCESTOR_DEPTH + 5 posts - to ensure pagination of ancestors
+ * The post will have MAX_CHILDREN_COUNT + 5 replies - to ensure pagination of children
+ * Each reply will have:
+ *  - If it is even indexed, it will have a number of replies equal to its index - this covers the case where a reply has no chain
+ *  - If it is odd indexed, it will have a reply chain of length equal to its index - this covers the case where a chain is paginated
+ */
+async function setupPosts(fixtureManager: FixtureManager) {
+    const [account] = await fixtureManager.createInternalAccount();
+
+    let postsToCreate = ReplyChainView.MAX_ANCESTOR_DEPTH + 6;
```

Comment:
+6 because one of these posts becomes _the_ post, so this creates the ancestors and the "current"